src/components/Forms/Select.js

```diff
@@ -0,0 +1,351 @@
+    /** DefaultOption */
+    defaultOption       : PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+        PropTypes.bool
+      ])
+    }),
+    /** Disable the text field */
+    disabled           : PropTypes.bool,
+    /** Text of label that will animate when TextField is focused */
+    floatingLabelText  : PropTypes.string,
+    /** Sets width to 100% */
+    fullWidth          : PropTypes.bool,
+    /** Sets width to 162px */
+    halfWidth          : PropTypes.bool,
+    /** FormComponent error for validation */
+    hasError           : PropTypes.bool,
+    /** Hint text will show up when the Select is open and there is no value */
+    hintText           : PropTypes.string,
+    /** Uniq id for Select */
+    id                 : PropTypes.string,
+    /** Handled by FormComponent after running built in validations */
+    isValid            : PropTypes.bool,
+    /** onOpen callback */
+    onOpen             : PropTypes.func,
+    /** onClose callback */
+    onClose            : PropTypes.func,
+    /** onFocus callback */
+    onFocus            : PropTypes.func,
+    /** onSelect callback returns option object {label: , value:}*/
+    onSelect           : PropTypes.func,
+    /** onBlur callback */
+    onBlur             : PropTypes.func,
+    /** Mark the field as required.  */
+    required           : PropTypes.bool,
+    /** Control the component selection manually */
+    selectedOption     : PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+        PropTypes.bool
+      ])
+    }),
+    /** Error from server to show ServerError message */
+    serverError        : PropTypes.string,
+    /** Wrapper styles, mainly used for custom width */
+    style              : PropTypes.object,
+    /** Text to show for validation error  */
+    validationErrorText: PropTypes.string,
+    /** Value will make TextField a controlled component  */
+  }
+
+  static defaultProps = {
+    disabled    : false,
+    defaultOption: null,
+    onFocus: () => {}, // eslint-disable-line no-empty-function
+    onBlur: () => {}, // eslint-disable-line no-empty-function
+    onOpen: () => {}, // eslint-disable-line no-empty-function
+    onClose: () => {}, // eslint-disable-line no-empty-function
+    onSelect: () => {} // eslint-disable-line no-empty-function
+  }
+
+  state = {
+    isOpen: false,
+    selectedOption: this.props.defaultOption || this.props.selectedOption || null
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { selectedOption } = nextProps
+    if (typeof selectedOption !== 'undefined' && selectedOption !== this.state.selectedOption) {
+      this.setState({selectedOption: selectedOption})
+    }
+  }
+
+  getValue = () => {
+    return (this.state.selectedOption && this.state.selectedOption.value) || null
+  }
+
+  _handleClose = () => {
```

Comment:
Why the `_` convention here? I think the prefix `handle` implies enough that this is a function within the component.